tests/integration/test_enrollment/test_agent_auth_enrollment.py

```diff
@@ -0,0 +1,76 @@
+'''
+brief: This module verifies the correct behavior of the agent-auth enrollment tool under different configurations
+copyright:
+    Copyright (C) 2015-2021, Wazuh Inc.
+    Created by Wazuh, Inc. <info@wazuh.com>.
+    This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+metadata:
+    component:
+        - Agent
+    modules:
+        - agent-auth
+    daemons:
+        - agent-auth
+    operating_system:
+        - Ubuntu
+        - CentOS
+        - Windows
+    tiers:
+        - 0
+    tags:
+        - Enrollment
+        - Agent-auth
+'''
+
+import pytest
+
+from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.services import control_service
+from conftest import *
+
+# Marks
+pytestmark = [pytest.mark.linux, pytest.mark.win32, pytest.mark.tier(level=0), pytest.mark.agent]
+
+configurations = load_wazuh_configurations(configurations_path, __name__)
+
+# Fixtures
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module"""
+    return request.param
+
+
+@pytest.mark.parametrize('test_case', [case for case in tests])
+def test_agent_auth_enrollment(set_test_case, configure_socket_listener, configure_environment, create_certificates,
+                               set_keys, set_pass, test_case: list):
```

Comment:
```suggestion
def test_agent_auth_enrollment(get_configuration, configure_socket_listener, configure_environment, 
                                                    create_certificates, set_keys, set_pass, test_case):
```

If we use `get_configuration` there is no need to use `set_test_case`

Reply:
Same as before regarding the set of a global variable with the current test-case for sage into a callback method.